Bottom: 17388aee0e0c32fe96a5d493e9c8a5a36574423e
Top:    7aae9d93afbe56b481dcd692ebb558eb4364851a
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-07-03 23:37:08 +0100

Refresh of avoid-walking-off-the-layer

---

diff --git a/src/misc.c b/src/misc.c
index f4ddcae..09fcdbb 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1045,7 +1045,7 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
   /* loop over all groups */
   for (s = group_string, group = 0;
        s && *s && group < *LayerN;
-       s++, group++)
+       group++)
     {
       while (*s && isspace ((int) *s))
         s++;
@@ -1092,6 +1092,8 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
             break;
         }
       LayerGroup->Number[group] = member;
+      if (*s == ':')
+        s++;
     }
 
   /* If no explicit solder or component layer group was found in the layer
